// pages/article/createArticle.js
import React, { useState, useEffect, useContext } from 'react';
import { Redirect } from 'react-router-dom'
import ArticleForm from 'components/articleForm'
import useFetch from 'hooks/useFetch'
import { CurrentUserContext } from 'contexts/currentUser'

const CreateArticle = () => {
    const [isSuccessfulSubmit, setIsSuccessfulSubmit] = useState(false)
    const [{ response, error }, doFetch] = useFetch('articles');
    const [currentUserState] = useContext(CurrentUserContext)
    const onSubmit = (article) => {
        // console.log('article', article)
        // console.log('currentUserState', currentUserState)
        doFetch({
            method: 'post',
            data: {
                article
            }
        })
    }

    const intialValues = {
        title: '',
        description: '',
        body: '',
        tagList: '',
    }

    useEffect(() => {
        if (!response) {
            return
        }
        setIsSuccessfulSubmit(true)
    }, [response])

    if (currentUserState.isLoggedIn === null) {
        return null
    }

    if (isSuccessfulSubmit || currentUserState.isLoggedIn === false) {
        return <Redirect to="/" />
    }

    return (<div><ArticleForm
        errors={(error && error.errors) || {}}
        intialValues={intialValues}
        onSubmit={onSubmit} /></div>);
}

export default CreateArticle;
