// contexts/currentUser.js
import React, {createContext, useState} from 'react'

// context is some data which we can access from every component 
// so this our first context which current user context 
// it can created using creactContext function of react which take array of 2 elements which initial values like empty object and function 
export const CurrentUserContext = createContext([{}, () => {}])

// this is the component which will wrap all our component which is called children
export const CurrentUserProvider = ({children}) => {

    const [state, setState] = useState({
        isLoading: false,
        isLoggedIn: null,
        currentUser: null
    })


    // so here we have initialize our state object with some property which we pass below to all our children component and from where we can set state using setState hook which we are passing from below 
    return (
        // here also we can understand Provider as kind of property on createContext function as we create object out of it which is CurrentUserContext so we call it as below 
        <CurrentUserContext.Provider value={[state, setState]}>
            {children}
        </CurrentUserContext.Provider>
    )
}
